micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java

```diff
@@ -161,7 +161,7 @@ public TimedAspect(MeterRegistry registry, Function<ProceedingJoinPoint, Iterabl
         this.shouldSkip = shouldSkip;
     }
 
-    @Around("@within(io.micrometer.core.annotation.Timed) && !@annotation(io.micrometer.core.annotation.Timed)")
+    @Around("@within(io.micrometer.core.annotation.Timed) && !@annotation(io.micrometer.core.annotation.Timed) && execution(* *(..))")
```

Comment:
The added part matches only method signatures (with a return type) so it will not apply to constructors that caused the runtime failures.